server/middleware/serverError.ts

```diff
@@ -0,0 +1,45 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import { ValidationError } from 'joi';
+import { JsonWebTokenError } from 'jsonwebtoken';
+import { CustomError } from '../utils';
+import { NextFunction, Request, Response } from 'express';
+
+const serverError = (
+  err: unknown,
+  req: Request,
+  res: Response,
+  next: NextFunction,
+) => {
+  if (err instanceof ValidationError) {
```

Comment:
When you throw a new CustomError each time you face an error in your controller, you don't need these checks.
You have to decide on a paradigm to use with your team.